src/sage/rings/burnside.py

```diff
@@ -0,0 +1,229 @@
+
+    def __le__(self, other):
+        return libgap.eval('fail') != libgap.ContainedConjugates(self.parent()._G, other._C, self._C, True)
+
+    def __eq__(self, other):
+        return _is_conjugate(self.parent()._G, self._C, other._C)
+
+class ConjugacyClassesOfSubgroups(Parent):
+    def __init__(self, G):
+        self._G = G
+        self._cache = dict() # invariant to subgroups
+        self._names = dict() # dictionary mapping subgroups to names
+        Parent.__init__(self, category=FiniteEnumeratedSets())
+
+    def _group_invariant(self, H):
+        return H.order()
+
+    def _normalize(self, H, name=None):
+        if not H.is_subgroup(self._G):
+            raise ValueError(f"{H} is not a subgroup of {self._G}")
+        p = self._group_invariant(H)
+        if p in self._cache:
+            for H0 in self._cache[p]:
+                if _is_conjugate(self._G, H, H0):
+                    if name is not None:
+                        self._names[H0] = name
+                    return H0
+            else:
+                self._cache[p].append(H)
+        else:
+            self._cache[p] = [H]
+        self._names[H] = name
+        return H
+
+    def _element_constructor_(self, x):
+        if x.is_subgroup(self._G):
+            return self.element_class(self, self._normalize(x))
+        raise ValueError(f"unable to convert {x} into {self}: not a subgroup of " + repr(self._G))
+
+    def set_name(self, H, name):
+        r"""
+        Rename conjugacy class of ``H`` to ``name``.
+        Passing ``None`` to ``name`` will remove any previously assigned name.
+        """
+        if not isinstance(name, str):
+            raise TypeError("name must be a string")
+        H_norm = self._normalize(H)
+        self._names[H_norm] = name
+
+    def __iter__(self):
+        return iter(self(H) for H in self._G.conjugacy_classes_subgroups())
+
+    def __contains__(self, H):
+        return H.is_subgroup(self._G)
+
+    def _repr_(self):
+        r"""
+        Return a string representation of ``self``.
+        """
+        return "Conjugacy classes of subgroups of " + repr(self._G)
+
+    Element = ConjugacyClassOfSubgroups
+
+class BurnsideRing(CombinatorialFreeModule):
+    def __init__(self, G, base_ring=ZZ):
+        """
+            sage: G = SymmetricGroup(4)
+            sage: B = BurnsideRing(G)
+            sage: TestSuite(B).run()
+        """
+        self._G = G
+        basis = ConjugacyClassesOfSubgroups(G)
+        basis.set_name(G, "1")
+        category = Algebras(base_ring).Commutative().WithBasis()
+        CombinatorialFreeModule.__init__(self, base_ring, basis, category=category)
+
+    def __getitem__(self, H):
+        r"""
+        Return the basis element indexed by ``H``. ``H`` must be a subgroup of ``self._G``.
+        """
+        if H.is_subgroup(self._G):
+            return self._from_dict({self._indices(H): 1})
+        else:
+            raise ValueError(f"{H} must be a subgroup of {self._G}")
```

Comment:
```suggestion

        raise ValueError(f"{H} must be a subgroup of {self._G}")
```

Reply:
I removed the error checking here entirely because it gets handled by `ConjugacyClassesOfSubgroups._element_constructor_` when we call `self._indices`.